### PR TITLE
Core 1695 delay

### PR DESCRIPTION
**Pull request by afakihcpr**
FYI @ayrton04, @efernandez, @servos

### CODE REVIEW

File: src/ros_filter.cpp [hunk 1]
```diff
@@ -383,6 +397,120 @@ namespace RobotLocalization
...
 
   template<typename T>
+  void RosFilter<T>::publishState(const ros::Time)
+  {
+      // Get latest state and publish it
+      nav_msgs::Odometry filteredPosition;
+
+      if (getFilteredOdometryMessage(filteredPosition))
+      {
+        worldBaseLinkTransMsg_.header.stamp = filteredPosition.header.stamp + tfTimeOffset_;
```
Comment (efernandez) [inline src/ros_filter.cpp]: It seems that most of this is only used if the first if condition is met; if not, it's overwritten, so IMO it should go inside the if clause.

File: src/ros_filter.cpp [hunk 2]
```diff
@@ -383,6 +397,120 @@ namespace RobotLocalization
...
+        positionPub_.publish(filteredPosition);
+      }
+  }
+
+  template<typename T>
+  void RosFilter<T>::integrationLoop()
+  {
+    // Not quite sure how to set the limits on the frequency in this case
+    // Set up the frequency diagnostic
+    double minFrequency = frequency_ - 20;
```
Comment (efernandez) [inline src/ros_filter.cpp]: Why `20`?

**Comment by efernandez** [comment]
LGTM in general

**Comment by ayrton04** [comment]
Overall, it LGTM. If this is just meant for CPR usage, then I don't see any need to change, but if we want to submit a PR upstream, then it would be good to make sure the node's default behavior doesn't change. There may be users who rely on a fixed update rate based on the `frequency` parameter.

**Comment by servos** [comment]
Make sure once this is merged that you talk to Jonathan (or Enrique knows how to do it) to get our buildbot to start building a deb of our fork instead of upstream.

**Comment by servos** [comment]
LGTM

**Pull request merged**